core/src/main/java/io/grpc/internal/CensusStatsModule.java

```diff
@@ -389,6 +392,7 @@ void callEnded(Status status) {
         tracer = BLANK_CLIENT_TRACER;
       }
       MeasureMap measureMap = module.statsRecorder.newMeasureMap()
+          // TODO(songya): remove the deprecated measure constants once they are completed removed.
```

Comment:
How is the migration handled on the consumer side in the Census implementation? For not to break the interoperability between gRPC and Census inside Google, it seems necessary for Census to read both the old and the new measurements, giving the new ones precedence. If that's the case, we could just remove the old measurements here.

Reply:
Unfortunately, inside Google the new measurements are not completely consistent with Census Java, and only the old measurements will be read at this moment. Even for open source, users could still export and read the old measurements so I think it's important to continue recording stats for the old ones.

There are some ongoing discussion about making the new measurements consistent with Census Java. /cc @dinooliva

Reply:
The artifact providing the gRPC views has [separate methods for registering the old and new views](https://github.com/census-instrumentation/opencensus-java/blob/07ad48d7c3031efcdd6ac3c8932cabe6ac3fff5b/contrib/grpc_metrics/src/main/java/io/opencensus/contrib/grpc/metrics/RpcViews.java#L130-L152), so users can choose when to upgrade them.  It might be easier for users to upgrade if they could get data for both sets of views at the same time.

Reply:
Is it possible to first change Census impls to convert the old measurements to the new ones, then switch gRPC to record the new measurements after all your customers have switched to the new impl and the new measurements? Then the migration of the measurements can go in line with the migration from NetworkEvent to MessageEvent (see my comment on that). Plus the gRPC code (esp. the tests) would be much cleaner if it ever has to deal with one API.

Reply:
I'm not sure, but I thought that we wanted users to explicitly switch to the new views, since the change isn't transparent like the switch to MessageEvent.  For example, the stats may be exported to a service that uses the view or measure names.

/cc @bogdandrutu

Reply:
Yes, like @sebright said, OpenCensus users could be exporting the old RPC constants to other backends and using them for alerting rules. Since the new and old RPC constants have different names, they will be different metrics on other monitoring systems (Stackdriver, Prometheus). We would like to allow users to explicitly choose old or new RPC constants and update their monitoring accordingly, as in our APIs ([registerAllGrpcViews](https://github.com/census-instrumentation/opencensus-java/blob/master/contrib/grpc_metrics/src/main/java/io/opencensus/contrib/grpc/metrics/RpcViews.java#L135) vs [registerAllViews](https://github.com/census-instrumentation/opencensus-java/blob/master/contrib/grpc_metrics/src/main/java/io/opencensus/contrib/grpc/metrics/RpcViews.java#L195)). So I think it's import to have stats for both of them.

Reply:
On top of that, as I mentioned in the previous comments, currently it's particularly difficult to map the old RPC constants to the new ones inside Google. For example, there are no *_count measures in the new constants.

Reply:
I see you don't have the "started count" measurements in the new constants, but it is a required metric inside Google.

Reply:
We removed started_count, finished_count and error_count in the new [specs](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md). There are some rationale behind why [error_count](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md#why-no-error-counts) and [finished_count](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md#why-are-completed_rpcs-views-defined-over-latency-measures) gets removed, but I cannot remember why we removed started_count.

@bogdandrutu @ramonza should we consider putting back the started_count measure since it's required?

Reply:
I added *_started_rpcs constants to OpenCensus `v0.14.0`, and upgraded the version here. Now the started_rpcs are recorded. PTAL at the two latest commits.